tools/matchcompiler.py

```diff
@@ -712,6 +719,10 @@ def convertFile(self, srcname, destname, line_directive):
         for function in self._rawMatchFunctions:
             strFunctions += function
 
+        constants = u''
+        for constString, constStringId in self._constStringCache.items():
+            constants += 'static const auto '  + constStringId + ' = MatchCompiler::makeConstString("' + constString + '");\n'
```

Comment:
This `auto` is `MatchCompiler::ConstString<n> where `n` is the length of the string, but to get the length correctly, we'd have to parse it (to get escaped characters correctly). This can certainly be done but I figured it wasn't worth it. The other option is to use `sizeof`, like `MatchCompiler::ConstString<sizeof(constString)>` but I didn't think that was much clearer.